rfcs/0003-storage.md

```diff
@@ -1,46 +1,108 @@
+* Validators will randomly sample the set of storage proofs to the set that they can handle, and only the creators of those chosen proofs will be rewarded.
+
+# 6. Validation and Replication Protocol
+
+## 6.1 Definitions
+
+1. Replicator: storage mining client, stores some part of the ledger enumerated in blocks and submits storage proofs to the chain. Not a full-node.
+2. Block: portion of the ledger which is downloaded by the storage client where storage proof data is derived.
+3. Storage proof: a set of sha hash state which is constructed by sampling the encrypted version of the stored ledger block at certain offsets.
+4. Fake storage proof: A proof which has the same format as a storage proof, but the sha state is actually from hashing a known ledger value which the storage
+client can reveal and is also easily verifiable by the network on-chain.
+5. Storage proof confirmation: A transaction by a validator which indicates the set of real and fake proofs submitted by a storage miner.
+6. Storage proof challenge: A transaction from a replicator that verifiably proves that a validator confirmed a fake proof.
+7. Storage proof claim: A transaction from a validator which is after the timeout period given from the storage proof confirmation and which no successful challenges have been observed which rewards the parties of the storage proofs and confirmations.
+8. Storage mining capacity: the number of keys and samples that a storage miner can verify each storage epoch.
+
+## 6.2 Constants
+
+1. NUM\_STORAGE\_ENTRIES: Number of entries in a block of ledger data. The unit of storage for a replicator.
+3. NUM\_KEY\_ROTATION\_TICKS: Number of ticks to save a PoH value and cause a key generation for the section of ledger just generated and the rotation of another key in the set.
+4. NUM\_STORAGE\_PROOFS: Number of storage proofs required for a storage proof claim to be successfully rewarded.
+5. RATIO\_OF\_FAKE\_PROOFS: Ratio of fake proofs to real proofs that a storage mining proof claim has to contain to be valid for a reward.
+6. NUM\_STORAGE\_SAMPLES: Number of samples required for a storage mining proof.
+7. NUM\_CHACHA\_ROUNDS: Number of encryption rounds performed to generate encrypted state.
+
+## 6.3 Validator behavior
+
+1. Validator joins the network and submits a storage mining capacity transaction which tells the network how many proofs it can process in a given period defined by NUM\_KEY\_ROTATION\_TICKS.
+2. Every NUM\_KEY\_ROTATION\_TICKS the validator stores the PoH value at that height.
+4. Every NUM\_KEY\_ROTATION\_TICKS it also validates samples received from replicators. It signs the PoH hash at that point and uses the low 5 bits of the first byte to index into the signature and then picks a number of bytes of the received sha states with that one in choosing which storage proofs to validate. It then creates a mask of valid proofs and fake proofs and sends it to the leader. This is a storage proof confirmation transaction.
```

Comment:
"and then picks a number of bytes of the received sha states with that one in choosing which storage proofs to validate." ?